src/Tizen.NUI/src/public/BaseComponents/DirectRenderingGLView.cs

```diff
@@ -0,0 +1,258 @@
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class DirectRenderingGLView : View
+    {
+        /// <summary>
+        /// The parameter of the RenderFrame Callback.
+        /// It has data to render directly.
+        /// </summary>
+        [SuppressMessage("Microsoft.Design", "CA1034:NestedTypesShouldNotBeVisible")]
+        public class RenderCallbackInput
+        {
+            Matrix mvp;
+            Matrix projection;
+            Size2D size;
+            Rectangle clippingBox;
+            ReadOnlyCollection<int> textureBindings;
+
+            public RenderCallbackInput(Matrix mvp, Matrix projection, Size2D size, Rectangle clippingBox, int[] textureBindings)
+            {
+                this.mvp = mvp;
+                this.projection = projection;
+                this.size = size;
+                this.clippingBox = clippingBox;
+                this.textureBindings = new ReadOnlyCollection<int>(textureBindings);
+            }
+
+            /// <summary>
+            /// MVP matrix
+            /// </summary>
+            public Matrix Mvp
+            {
+                get { return mvp; }
+            }
+
+            /// <summary>
+            /// Projection matrix
+            /// </summary>
+            public Matrix Projection
+            {
+                get { return projection; }
+            }
+
+            /// <summary>
+            /// The size of the DirectRenderingGLView
+            /// </summary>
+            public Size2D Size
+            {
+                get { return size; }
+            }
+
+            /// <summary>
+            /// The area of DirectRenderingGLView. You can use this for glScissor()
+            /// </summary>
+            public Rectangle ClippingBox
+            {
+                get { return clippingBox; }
+            }
+
+            /// <summary>
+            /// Texture bindings
+            /// </summary>
+            public ReadOnlyCollection<int> TextureBindings
+            {
+                get { return textureBindings; }
+            }
+        }
+
+        private GLInitializeDelegate glInitializeCallback;
+        private GLRenderFrameDelegate glRenderFrameCallback;
+        private GLTerminateDelegate glTerminateCallback;
+        private InternalGLRenderFrameDelegate internalRenderFrameCallback;
+
+        /// <summary>
+        /// Type of callback to initialize OpenGLES.
+        /// </summary>
+        public delegate void GLInitializeDelegate();
+
+        /// <summary>
+        /// Type of callback to render the frame with OpenGLES APIs.
+        /// If the return value of this callback is not 0, the eglSwapBuffers() will be called.
+        /// </summary>
+        /// <returns>The return value is not 0, the eglSwapBuffers() will be called.</returns>
+        public delegate int GLRenderFrameDelegate(in RenderCallbackInput input);
+
+        private delegate int InternalGLRenderFrameDelegate(global::System.IntPtr cPtr);
```

Comment:
This looks wrong order.
Currently this private delegate is between two public delegate.